packages/components/src/internal/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -725,6 +732,17 @@ class SampleTypeDesignerImpl extends React.PureComponent<Props & InjectedBaseDom
                     namePreviewsLoading={namePreviewsLoading}
                     namePreviews={namePreviews}
                     onNameFieldHover={this.onNameFieldHover}
+                    nameExpressionGenIdProps={
+                        showGenIdBanner && options
+                            ? {
+                                  dataTypeName: options.get('name'),
+                                  dataTypeLSID: options.get('lsid'),
+                                  rowId: options.get('rowId'),
+                                  kindName: 'SampleSet',
```

Comment:
I'm wondering if we want constants or an enum or a declared type for 'SampleSet' and 'DataClass'.  We use these strings in other places too, so I can see this is just following that trend.

Reply:
I initially used enum, but the problem is typescript is not allowing defining a type to be : ENUM_A | ENUM_B , it's only accepting string_a | string_b

Reply:
Ah right.  I remember that the treatment of enums is, well, not ideal.